NEXT-25815 - Recurring payments (#982)
shopware/docs: guides/plugins/plugins/checkout/payment/add-payment-plugin.md

You can create your own payment handler by implementing one of the following interfaces:

| Interface                           | DI container tag                    | Usage                                                                                              |
|:------------------------------------|:------------------------------------|:---------------------------------------------------------------------------------------------------|
| SynchronousPaymentHandlerInterface  | `shopware.payment.method.sync`      | Payment can be handled locally, e.g. pre-payment                                                   |
| AsynchronousPaymentHandlerInterface | `shopware.payment.method.async`     | A redirect to an external payment provider is required, e.g. PayPal                                |
| PreparedPaymentHandlerInterface     | `shopware.payment.method.prepared`  | The payment was prepared beforehand and will only be validated and captured by your implementation |
| RefundPaymentHandlerInterface       | `shopware.payment.method.refund`    | The payment allows refund handling                                                                 |

> | RecurringPaymentHandlerInterface    | `shopware.payment.method.recurring` | The payment allows recurring payments (e.g. subscriptions)                                         |

Reviewer: ```suggestion
| RecurringPaymentHandlerInterface    | `shopware.payment.method.recurring` | The payment allows recurring payments, e.g. subscriptions                                       |
```

* `capture`: This method will be called after an order was placed, but only if the validation did not fail and stop the payment flow before. At this point, the order was created and the payment handler will be called again to charge the payment. When the charge was successful, the payment handler should update the transaction state to `paid`. The user will be forwarded to the finish page. Throw an `CapturePreparedPaymentException` on any errors to fail the capture process and the after order process will be active, so the customer can complete the payment again.
* `refund`: This method is called, whenever a successful transaction is claimed to be refunded. The implementation of the refund handler should validate the legitimacy of the refund and call the PSP to refund the given transaction. Throw a `RefundException` to let the refund fail.

> * `captureRecurring`: This method is called, whenever a recurring payment is charged. At this point, a valid billing agreement with the payment provider should exist. Use some of the other payment methods for handling the initial order and billing agreement. Use this interface only for handling all recurring captures afterwards.

Reviewer: ```suggestion
* `captureRecurring`: This method is called whenever a recurring payment is charged. At this point, a valid billing agreement with the payment provider should exist. Use some of the other payment methods for handling the initial order and billing agreement. Use this interface only for handling all recurring captures afterwards.
```

> All methods - bar the `captureRecurring` - get the `\Shopware\Core\System\SalesChannel\SalesChannelContext` injected. Please note, that this class contains properties, which are nullable. If you want to use this information, you have to ensure in your code that they are set and not `NULL`.

Reviewer: ```suggestion
All payment handler methods have the `\Shopware\Core\System\SalesChannel\SalesChannelContext` injected, except for the new  `captureRecurring` . 
Note that this class contains properties that are nullable. If you want to use this information, you have to ensure in your code that they are set and not `NULL`.
```

> All methods - bar the `captureRecurring` - get the `\Shopware\Core\System\SalesChannel\SalesChannelContext` injected. Please note, that this class contains properties, which are nullable. If you want to use this information, you have to ensure in your code that they are set and not `NULL`.

Reviewer: @lernhart What is meant by "-bar the `captureRecurring` -" here?

Reviewer: @mstegmeyer, Can you please take a look at this?

Author: It means, that all payment handler methods (sync, async, etc.) have a salesChannelContext injected, except the new `captureRecurring`.

> All payment handler methods have the `\Shopware\Core\System\SalesChannel\SalesChannelContext` injected, except for the new  `captureRecurring`  method. Please note, that this class contains properties, which are nullable. If you want to use this information, you have to ensure in your code that they are set and not `NULL`.

Reviewer: ```suggestion
All payment handler methods have the `\Shopware\Core\System\SalesChannel\SalesChannelContext` injected, except for the new  `captureRecurring`  method. Note that this class contains properties that are nullable. If you want to use this information, you have to ensure in your code that they are set and not `NULL`.
```
